sno/merge.py

```diff
@@ -267,3 +247,69 @@ def output_merge_json_as_text(jdict):
         # TODO: explain how to resolve conflicts, when this is possible
         click.echo("Sorry, resolving merge conflicts is not yet supported", err=True)
         click.echo("Use `sno merge --abort` to abort this merge", err=True)
+
+
+@click.command()
+@click.option(
+    "--ff/--no-ff",
+    default=True,
+    help=(
+        "When the merge resolves as a fast-forward, only update the branch pointer, without creating a merge commit. "
+        "With --no-ff create a merge commit even when the merge resolves as a fast-forward."
+    ),
+)
+@click.option(
+    "--ff-only",
+    default=False,
+    is_flag=True,
+    help=(
+        "Refuse to merge and exit with a non-zero status unless the current HEAD is already up to date "
+        "or the merge can be resolved as a fast-forward."
+    ),
+)
+@click.option(
+    "--dry-run",
+    is_flag=True,
+    help="Don't perform a merge - just show what would be done",
+)
+@eager_callback_flag(
+    '--abort',
+    callback=abort_merging_state,
+    help="Abandon an ongoing merge, revert repository to the state before the merge began",
+)
+@eager_callback_flag(
+    '--continue',
+    callback=complete_merging_state,
+    help="Completes and commits a merge once all conflicts are resolved and leaves the merging state",
+)
+@click.argument("commit", required=True, metavar="COMMIT")
+@do_json_option
+@click.pass_context
+def merge(ctx, ff, ff_only, dry_run, do_json, commit):
```

Comment:
Should `merge` be a [Click group](https://click.palletsprojects.com/en/7.x/commands/) (like `sno` is)? 

(So `merge continue`/`merge abort`/`merge whatever` can have separate options eg: `--message` without needing complex argument-parsing machinations?)

Maybe that's later changes as it gets more complicated.

Reply:
IMO no:
* we don't need any continue/abort options (at the moment?)
* it'd make it hard/confusing/ambiguous to merge a branch called `continue` or `abort`, which is bound to come up sometimes
* without a good reason to change it, we should stick with the git usage: `git merge --continue`  / `git merge --abort`

Reply:
Leaving as-is for now